libnetwork/libnetwork_linux_test.go

```diff
@@ -1498,45 +1191,30 @@ func TestEndpointJoin(t *testing.T) {
 		libnetwork.NetworkOptionEnableIPv6(true),
 		libnetwork.NetworkOptionIpam(defaultipam.DriverName, "", nil, ipamV6ConfList, nil),
 		libnetwork.NetworkOptionDeferIPv6Alloc(true))
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, err)
 	defer func() {
-		if err := n1.Delete(); err != nil {
-			t.Fatal(err)
-		}
+		assert.Check(t, n1.Delete())
 	}()
 
 	ep1, err := n1.CreateEndpoint(context.Background(), "ep1")
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, err)
 	defer func() {
-		if err := ep1.Delete(context.Background(), false); err != nil {
-			t.Fatal(err)
-		}
+		assert.Check(t, ep1.Delete(context.Background(), false))
 	}()
 
 	// Validate if ep.Info() only gives me IP address info and not names and gateway during CreateEndpoint()
 	info := ep1.Info()
 	iface := info.Iface()
-	if iface.Address() != nil && iface.Address().IP.To4() == nil {
-		t.Fatalf("Invalid IP address returned: %v", iface.Address())
-	}
-	if iface.AddressIPv6() != nil && iface.AddressIPv6().IP == nil {
-		t.Fatalf("Invalid IPv6 address returned: %v", iface.Address())
-	}
-
-	if len(info.Gateway()) != 0 {
-		t.Fatalf("Expected empty gateway for an empty endpoint. Instead found a gateway: %v", info.Gateway())
+	if iface.Address() != nil {
+		assert.Check(t, iface.Address().IP.To4() != nil, "Invalid IP address returned: %v", iface.Address())
 	}
-	if len(info.GatewayIPv6()) != 0 {
-		t.Fatalf("Expected empty gateway for an empty ipv6 endpoint. Instead found a gateway: %v", info.GatewayIPv6())
+	if iface.AddressIPv6() != nil {
+		assert.Check(t, iface.AddressIPv6().IP != nil, "Invalid IPv6 address returned: %v", iface.AddressIPv6())
 	}
```

Comment:
The old code did the same - but could align this with the IPv4 check, and make sure it is IPv6? ...

```suggestion
	if iface.AddressIPv6() != nil {
		assert.Check(t, iface.AddressIPv6().IP.To6() != nil, "Invalid IPv6 address returned: %v", iface.AddressIPv6())
	}
```

Reply:
Hm.. there's no `To6()`.  There' a  `To16()`, but looking at that one again,  it would be  happy to return both IPv4 or  IPv6 https://github.com/golang/go/blob/608acff8479640b00c85371d91280b64f5ec9594/src/net/ip.go#L226-L236


So I guess we need to do a `To4()` and check that it's nil (i.e., "not and IPv4 address, so IPv6")

Reply:
Oh, sorry - yes.